src/makefun/main.py

```diff
@@ -960,15 +964,9 @@ def _get_args_for_wrapping(wrapped, new_sig, remove_args, prepend_args, append_a
 
     # attributes: start from the wrapped dict, add '__wrapped__' if needed, and override with all attrs.
     all_attrs = copy(getattr_partial_aware(wrapped, '__dict__'))
+    all_attrs.setdefault("__wrapped__", wrapped)
```

Comment:
Is there any reason why is `__wrapped__` set only if not already present (setdefault) while `__signature__` is overridden (line 969) ?

Reply:
I think you're right: it makes sense to update `__wrapped__` if not already present. It appears that functools.wraps does this (see below), and it makes sense. I will make this change here:

```suggestion
    all_attrs["__wrapped__"] = wrapped
```

We have to set `__signature__`, even if it's already present, to handle the case of applying two signature-changing decorators.

```python
>>> import functools
>>> def foo1():
...     pass
...
>>> @functools.wraps(foo1)
... def foo2():
...     pass
...
>>>
>>> @functools.wraps(foo2)
... def foo3():
...     pass
...
>>>
>>> assert foo3.__wrapped__ is foo2
>>> assert foo2.__wrapped__ is foo1
```

Reply:
This looks better indeed. Thanks!